dbt/include/decodable/macros/materializations/table/table.sql

```diff
@@ -29,7 +29,8 @@
   {% set should_create = {'value': false} %}
   {% if existing_relation is not none %}
     {% set watermark = config.get('watermark') %}
-    {% if adapter.has_changed(sql, target_relation, watermark) or should_full_refresh() %}
+    {% set primary_key = config.get('primary_key') %}
+    {% if adapter.has_changed(sql, target_relation, watermark, primary_key) or should_full_refresh() %}
```

Comment:
I think it is the right thing to do to consider the primary key field in this comparison (we also consider watermark, for example). However, I am a bit uncertain about the behavior in general: If changes to the target stream are detected, it is simple deleted and recreated. Is that intended behavior @gunnarmorling @rmetzger ?

Reply:
Recreation of changed streams is the desired behavior. I will document this in the README if it isn't there already.